Assets/VRTK/Scripts/VRTK_RealMovement.cs

```diff
@@ -0,0 +1,350 @@
+                    }
+                    float avg = sum / m_averagePeriod;
+
+                    // Add the average to the the list average.
+                    listAverage += avg;
+                }
+
+                float speed = ((m_speedScale * 350) * (listAverage / m_trackedObjects.Count));
+
+                if (speed > m_maxSpeed && m_maxSpeed >= 0)
+                    speed = m_maxSpeed;
+
+                m_direction = Vector3.zero;
+
+                if (m_dirMethod == DirectionalMethod.SmartDecoupling || m_dirMethod == DirectionalMethod.DumbDecoupling)
+                {
+                    if (m_initalGaze.Equals(Vector3.zero))
+                    {
+                        m_initalGaze = new Vector3(headset.forward.x, 0, headset.forward.z);
+                    }
+
+                    if (m_dirMethod == DirectionalMethod.SmartDecoupling)
+                    {
+                        bool closeEnough = true;
+                        float curXDir = headset.rotation.eulerAngles.y;
+                        if (curXDir <= m_smartDecoupleThreshold)
+                        {
+                            curXDir += 360;
+                        }
+                    
+                        closeEnough = closeEnough && (Mathf.Abs(curXDir - controllerLeftHand.transform.rotation.eulerAngles.y) <= m_smartDecoupleThreshold);
+                        closeEnough = closeEnough && (Mathf.Abs(curXDir - controllerRightHand.transform.rotation.eulerAngles.y) <= m_smartDecoupleThreshold);
+
+                        if (closeEnough)
+                            m_initalGaze = new Vector3(headset.forward.x, 0, headset.forward.z);
+                    }
+                    m_direction = m_initalGaze;
+                }
+                else if (m_dirMethod.Equals(DirectionalMethod.Gaze))
+                {
+                    m_direction = (new Vector3(headset.forward.x, 0, headset.forward.z));
+                }
+
+                m_curSpeed = speed;
+
+                m_cameraLastYRotation = m_direction.y;
+            }
+
+            foreach (Transform trackedObj in m_trackedObjects)
+            {
+                // Get delta postions and rotations
+                m_previousYPositions[trackedObj] = trackedObj.transform.localPosition.y;
+            }
+
+            Vector3 movement = (m_direction * m_curSpeed) * Time.fixedDeltaTime;
+
+            /*if (m_playerBody != null)
+            {
+                m_playerBody.updatePlayerBodyPosition();
+                Vector3 cameraDistance = m_playerBody.getCameraDistance();
+                movement = movement + m_playerBody.getImmersiveMovementModifier() - (cameraDistance);
+                m_playerBody.resetImmersiveMovementModifier();
+            }*/
+
+            m_rigidBody.MovePosition(m_rigidBody.transform.position + movement);
+
+            m_playAreaPreviousY = transform.position.y;
+        }
+
+        public Vector3 getMovementDirection()
```

Comment:
Public methods should be above private and protected methods. Also `GetMovementDirection()` (uppercase start)